kafka-client/src/main/java/dev/responsive/kafka/store/FactSchemaWriter.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2023 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.responsive.kafka.store;
+
+import com.datastax.oss.driver.api.core.cql.Statement;
+import dev.responsive.db.CassandraClient;
+import dev.responsive.db.RemoteSchema;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+public class FactSchemaWriter<K> implements RemoteWriter<K> {
+
+  private final CassandraClient client;
+  private final RemoteSchema<K> schema;
+  private final String name;
+  private final int partition;
+
+  private final List<Statement<?>> statements;
+
+  public FactSchemaWriter(
+      final CassandraClient client,
+      final RemoteSchema<K> schema,
+      final String name,
+      final int partition
+  ) {
+    this.client = client;
+    this.schema = schema;
+    this.name = name;
+    this.partition = partition;
+
+    statements = new ArrayList<>();
+  }
+
+  @Override
+  public void insert(final K key, final byte[] value) {
+    statements.add(schema.insert(name, partition, key, value));
+  }
+
+  @Override
+  public void delete(final K key) {
+    statements.add(schema.delete(name, partition, key));
+  }
+
+  @Override
+  public CompletionStage<RemoteWriteResult> flush() {
+    final List<CompletionStage<RemoteWriteResult>> results = statements.stream()
+        .map(this::executeAsync)
```

Comment:
This doesn't guarantee any ordering right? I think this is safe for two reasons:
1. in the current store implementation, we'll never get two writes for the same key
2. even if that was not the case, the fact schema assumes that there are no updates
Might want to leave a comment clarifying that.

Reply:
correct, but also the commit buffer will never contain two values for the same key so this would be safe for any implementation. i'll add a comment